Tasks/DeployAzureResourceGroup/Utility.ps1

```diff
@@ -487,6 +488,26 @@ function Instantiate-Environment
     Write-Verbose "Added the environmnent $outputVariable to output variable" -Verbose
 }
 
+function Get-AzureResourcesTags
+{
+    param($resources,
+          $azureVmDetails)
+
+    $tagsList = New-Object 'system.collections.generic.dictionary[[string],[system.collections.generic.dictionary[[string],[string]]]]' 
+
+    foreach($resource in $resources)
+    {
+        $resourceName = $resource.Name
+        if(-not ($resource.Tags -eq $null or $resource.Tags.Count -eq 0))
+        {
+            $resourceFqdn = ($azureVmDetails[$resourceName]).fqdn
+            $tagsList.Add($resourceFqdn, $resource.Tags)
```

Comment:
resource fqdn is not unique we should use resource name as a key to get tags corresponding to that resource